core/sequencer/runner/native.go

```diff
@@ -152,26 +158,37 @@ type ReduceMutationFn func(msgs []*pb.EntryUpdate, leaves []*pb.EntryUpdate,
 
 // DoReduceFn takes the set of mutations and applies them to given leaves.
 // Returns a list of key value pairs that should be written to the map.
-func DoReduceFn(reduceFn ReduceMutationFn, joined []*Joined, emitErr func(error),
-	incFn IncMetricFn) []*entry.IndexedValue {
-	ret := make([]*entry.IndexedValue, 0, len(joined))
-	for _, j := range joined {
-		incFn("ReduceFn")
-		reduceFn(j.Values1, j.Values2,
-			func(e *pb.EntryUpdate) {
-				ret = append(ret, &entry.IndexedValue{Index: j.Index, Value: e})
-			},
-			wrapErrFn(emitErr, fmt.Sprintf("reduceFn on index %x", j.Index)),
-		)
-	}
+func DoReduceFn(reduceFn ReduceMutationFn, joined <-chan *Joined, emitErr func(error),
+	incFn IncMetricFn) <-chan *entry.IndexedValue {
+	ret := make(chan *entry.IndexedValue)
+	go func() {
+		defer close(ret)
+		var wg sync.WaitGroup
+		defer wg.Wait()
+		for w := 1; w < runtime.NumCPU(); w++ {
```

Comment:
I feel like there should be some configurable parameter somewhere which defaults to `NumCPU`. Maybe leave a TODO for that?